zipline/pipeline/data/dataset.py

```diff
@@ -153,7 +166,18 @@ def __new__(cls,
                 name,
                 doc,
                 metadata,
-                currency_conversion):
+                currency_conversion,
+                currency_aware):
+        if currency_aware and dtype != float64_dtype:
+            raise ValueError(
+                '{} cannot be constructed with currency_aware={}, dtype={}. '
```

Comment:
Might be useful to include the dataset in this message too. `name` here will just be the column name, so if there are multiple datasets with similar column names, the name won't be helpful.

Reply:
We should probably also add a quick test case for this.